<!DOCTYPE html>
<html lang="en">

<head>

    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, shrink-to-fit=no, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="">

    <title>Gene Regulatory Network</title>
    <!-- Bootstrap Core CSS -->
    <link href="css/bootstrap.min.css" rel="stylesheet">

    <!-- Custom CSS -->
    <link href="css/simple-sidebar.css" rel="stylesheet">

</head>
<body>
        <!-- Page Content -->
        <div id="page-content-wrapper">
            <div class="container-fluid" style="">
              <header style="">
              <h3> SC462 Elements of Synthetic Biology Spring 2017 </h3>
              <h1 > Gene Regulatory Networks </h1>
            </header>
              <nav class="navbar navbar-default navbar-fixed-top">
  <div class="container"><ul class="nav navbar-nav nav-tabs  nav-justified">
     <li class="active"><a href="#ppt">Project Presentation</a></li>
     <li><a href="#report">Project Report</a></li>
     <li><a href="#source">Source Code</a></li>
     <li><a href="#links">Video Links</a></ul>

  </div>
  </nav>
    <div class="container">
                <div class="row" id="ppt">
                    <div class="col-lg-12" >
                        <h1>Project Presentation</h1>
                        <iframe src="https://docs.google.com/presentation/d/1zsWztE2hZ6IrDa_X6Z3UNm1L0VKYgLTa6UgAaRiYYl0/embed?start=false&loop=false&delayms=3000"
                        frameborder="0" width="900" height="500" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true" style="margin:auto"></iframe>
                    </div>
                </div>
                <div class="row" id="report">
                    <div class="col-lg-12" >
                        <h1>Project Report</h1>
                        <h3>Introduction</h3>
                        <p>Gene Regulation Gene regulation is the mechanism by which the production
of specic gene products(protein nor RNA) is increased or decreased in a cell.
Developmental pathways can be triggered by modifying the gene expression pat-
terns of the cells. All points of gene expression can be regulated. Transcription,
RNA processing and transport, translation and post-translational modication
of a protein, and mRNA degradation are the parts of gene regulation. Regula-
tion of Transcription: A cell determines when a gene is transcribed by regulating
transcription. A cell increases or decreases amount of mRNA generated from
each gene. There are ve dierent ways that RNA polymerase (the enzyme
that makes RNA from a DNA template) can be aected: 1. The specicity of
RNA polymerase for promoters can be changed by a cell using specicity fac-
tors. These regulatory proteins will make RNA polymerase more or less likely
to bind to a promoter. 2. Repressor proteins can bind to an upstream coding
sequence, known an operator, to inhibit gene transcribing by RNA polymerase..
This leads to decreased expression of a gene. 3. Activator proteins can bind
to the operator region to increase the anity of RNA polymerase towards the
promoter. 4. Regulatory regions that lie far upstream to their target gene are
called enhancer regions. Activator binding to an enhancer leads the DNA to
form a loop, which puts the enhancer next to its target gene. This creates in-
creased expression of the target gene. 5. To completely turn o expression of
a gene, cell uses silencer regions. The gene is silenced when a silence region is
bound by transcription factors. This implies that there is no expression of the
gene.
</p>
<h3> Gene Regulation in Prokaryotes</h3>
<p>[5]Regulatory proteins carry the signals to the genes. Regulatory proteins are of
two types. They are positive regulators called activators and negative regulators
called repressors. These activators and repressors are DNA binding proteins.
N
egative Regulators or Repressors The repressor or inhibitor protein binds to
the operator on DNA. The RNA polymerase enzyme is blocked by the repressor
from binding to the promoter which prevents transcription. The repressor binds
to the site and overlaps the polymerase enzyme. This turns o the activity of
the genes. This is called negative control mechanism. An anti-inhibitor or anti-
repressor called inducer inactivates the repressor and thus activates the genes.
This is demonstrated by lactose operon
P
ositive Regulators or Activators The activator allows polymerase enzyme to
bind to the promoter thereby causing activation of transcription by the pro
moter.</p>
<img src="" />
<h4>2.1 Operon</h4>
<p>In bacteria cistrons or structural genes produce enzymes of a metabolic path-
way. These are organised in a cluster which have related fucntions. Poly-
cistronic genes of prokaryotes plus their regulatory genes constitute a system
called operon. Operon is a unit of expression and regulation.</p>
<p>
  <h5>1.Lactose Operon or Lac Operon</h5>
This is a negative control mechanism. The synthesis of enzyme (3-galactosidase
has bee5n studied in detail. This enzyme breaks down lactose into glucose and
galactose. -galactosidase is present in minimal amounts in the absence of lactose.
As soon as lactose is added from outside, the production of - galactosidase is in-
creased by thousand times. As soon as the lactose is consumed, the production
of the enzyme drops again. The enzymes whose production can be increased
by the presence of the substrate are called inducible enzymes. When lactose
is added to the culture medium of E. coli, it induces the formation of three
enzymes (5-galactosidase, permease and transacetylase, which breaks down lac-
tose into galactose and glucose. The genes, which code for these enzymes form
a cluster and are called cistrons or structural genes. They are transcribed at the
same time into a single mRNA chain, which has codons corresponding to all the
three enzymes. The mRNA transcribed from many genes is called polycistronic.
Regulatory genes control the functioning of structural genes to produce mRNA.
There are three structural genes Z, Y and A, which code for enzymes p-galactosidase,
lac permease and transacetylase respectively. Regulatory genes has Regulator I,
a control gene called operator gene O and promoter P. I gene produces a protein
called repressor or inhibitor. The repressor being active, binds to the operator
O and switches o the operator. Thus, the transcription is stopped.
When the inducer (lactose) is supplied from outside, the inducers binds to the
repressor. The l vbactose changes into allolactase on entering the bacteria. Al-
lolactose changes the shape of the repressor which makes it inactive. It cannot
bind to the operator then. The operator becomes free and is \turned on" which
starts the transcription. The presence of the inducer allows the transcription
of Lac operon, as it is not blocked by the repressor protein. The synthesis of
enzymes due to the presence of specic substrate (lactose) is called induction.
It is also called de-repression.</p>
<p>
<h5>2.Tryptophan Operon</h5>
It is also a negative control system but it also forms a biosynthetic pathway.
It is known as repressible system. It works on the clause that when the amino
acid tryptophan is present, the tryptophan operon need not be activated.
The co-repressor activates the repressor protein and it binds the operator and
switches it \o'. Tryptophan is synthesized in ve steps, where each step re-
quires a specic enzyme. The genes which encodes these enzymes lie next to one
another, called trp B, trp D, trp E, trp C, trp A. Tryptophan operon codes for
ve enzymes for the synthesis of amino acid tryptophan. In repressible system,
the regulatory gene produces a repressor protein, which is normally inactive
and doesn't bind to operator on DNA. The repressor on joining the co-repressor
undergoes conformational changes that activate it and enable it to bind to the
operator. This doesn't allow RNA polymerase enzyme to bind to the promoter.
This is opposite to the situation of lac operon in which the repressor is active
on its own and when it binds to the inducer, loses the anity for the operator.
Operon is regulated by tryptophan and this represses the synthesis of tryp-
tophan. The end product of the metabolic chain stops the synthesis of en-
zymes.This allows the bacteria to synthesize enzymes only if needed. This is
known as feedback repression. In feedback inhibition the end product of a
metabolic pathway serves as an allosteric inhibitor of the rst enzyme of the
metabolic chain. Induction and repression prevent the synthesis of unnecessary
enzymes and save valuable energy.</p>
<h3>3. Gene Regulation in Eukaryotes</h3>
<p>
1).Stimulation through a signaling pathway (e.g. presence of an environmental
stimulation, followed by internal transduction of the signal)
2).Effective stimulation of a transcription factors
3).Activation of the transcription of a particular gene
4).Production of messenger RNA (mRNA)
5).Translation of mRNA, i.e. production of a functional protein.
Gene regulation is about which genes are controlled by which cells and out
of the many genes in its genome, which are "turned on". Each cell has a unique
set of active genes even when all the cells have exact same DNA. All the steps
in eukaryotic gene expression can be regulated. Genes are regulated at multiple
steps.    
</p>
<p>
<h5>Chromatin accessibility</h5>
The chromatin structure can be regulated. More open chromatin makes a gene
more available for transcription.
</p>
<p>
<h5>Transcription</h5>
Transcription is a key regulatory point for many genes. Sets of transcription fac-
tor proteins bind to specific DNA sequences in or near a gene and promote or re-
press its transcription into an RNA.
</p>
<p>
<h5>RNA processing</h5>
Splicing, capping, exit from nucleus and addition of a poly-A tail to an RNA
molecule can be regulated. Alternative splicing make different mRNAs from the
same.
</p>
<p>
<h5>Stages of eukaryotic gene expression</h5>
1).Chromatin structure. Chromatin may be tightly compacted or loose and
open.
2).Transcription. A primary transcript is made by an available gene by tran-
scription.
3).Processing and export. The primary transcript is processed (spliced, capped,
given a poly-A tail) and sent out of the nucleus.
mRNA stability. In the cytosol, the mRNA may be stable for long or may be
quickly degraded (broken down).
4).Translation. The mRNA may be translated by ribosomes to make a polypep-
tide.
5).Protein processing. The polypeptide may undergo various types of process-
ing, including proteolytic cleavage (snipping off of amino acids) and addition of
chemical modifications, such as phosphate groups.
For a given gene, all these steps need to be executed for an active protein to be
present in the cell.
</p>
<p>
<h5>RNA stability</h5>
How many proteins can be made from an mRNA molecule depends on the life-
time of an mRNA molecule. Small regulatory RNAs called miRNAs can bind
to target mRNAs and lead to their chopping off.
</p>
<p>
<h5>Translation</h5>
Translation of an mRNA may be increased or inhibited by regulators. For in-
stance, miRNAs sometimes block translation of their target mRNAs than their
chopping off.
</p>
<p>
<h5>Protein activity</h5>
Proteins go through a variety of modifications, such as being chopped up or
tagged with chemical groups. The activity of the protein may be affected by
these modifications.[3]
</p>
<p>
<h5>Chromatin accessibility</h5>
The chromatin structure can be regulated. More open chromatin makes a gene
more available for transcription.
</p>
<h3>4. Gene Regulatory Network</h3>
<p>
    Gene regulatory network (GRN) is a cumulation of molecular regulators that
interplay among themselves and with other materials in the cell to control the
gene expression levels of mRNA and proteins. The regulator can be DNA, RNA,
protein and complexes of these. The synergy can be direct or indirect . It can be
through transcribed RNA or either through a translated protein. Each mRNA
molecule goes on to make a particular protein. Sometimes, this protein will in-
corporate at the cell covering or within the cell to give it particular design prop-
erties. In other cases the protein will be an enzyme. Some proteins trigger other
genes, and these are the transcription factors that play the main role in regula-
tory networks.By binding to the promoter region at the start of other genes they
make them start initiating the production of another protein. Some transcrip-
tion factors are inhibitory.
The GRN is modelled as a directed graph; G = (V, E) , where V represents the
set of all nodes (genes) and E is the set of all edges (the interaction between a
pair of genes). An edge e in set E represents physical and/or regulatory inter-
actions between two genes.Genes can be viewed as nodes in the network, with
input being proteins such as transcription factors, and outputs being the level
of gene expression. The node itself can also be viewed as a function which can
be obtained by combining basic functions upon the inputs. Thus by modelling
gene regulatory network we get weight matrix giving the output of interactions
between two nodes. Now using these output weight matrix value we can cal-
culate the value of gene expression required for a particular change in a cell.[4]
</p>
<h3>5. Gene expression networks from microarray dataset</h3>
<p>
  There two types of dierent networks coupled with the gene expression. One
is gene expression networks. To derive these gene expression networks, we use
microarray data sets. On a genome scale, we take measures of time series of
mRNA levels. The other type of network is the transcription factor networks.
These networks are derived from a method which identies all the DNA binding
sites for all the transcription factors.
One of the issues in bioinformatics is to recognize the relation between all the
dierent parts of a system and deduce what impact these parts have on each
other. We need complete picture of gene regulatory network to comprehend the
system of gene expression.
These networks gives various characteristics of gene regulation and deliver dif-
ferent insights. The gene expression network can be gathered from analysis of
data based on time series of gene expression. All the resulting networks repre
sents features that are global to biological schemes.
Gene expression arrays which is a high output method, emphasis on genome
proles of a single gene from a population. We can say this represents a `picture
in time' of a diverse population. In the same species also genetic variations
aects expression prole of the gene. Hence two organisms are not the same.
Also, every organism contains a history of its own. This history can direct to an
extensive range of states of the system with various dierent expression proles.
We need to monitor the expression prole of the gene time to time to dieren-
tiate between consequences of the history and the generic variation. Analysing
and deciding the time series expression data has a lot of advantages. For ex-
ample, the chemical kinetics gives more autonomous data than the chemical
thermodynamics. In the same way, the time series data are more agreeable to
network modeling than expression data from a population. With the time series
data set, initially the system is made in a given physiological state and as it
changes the state, we measure the changes in the gene expression levels.
To analyze the expression time series we use random forest regression model.
</p>
<h3>6. Random Forest Regressor Algorithm</h3>
<p>
  Random forest Algorithm works as large collection of decorrelated trees. This
algorithm makes use of lots of decision trees to make a classication. These de-
cision trees are developed on the basis of random selection of data and random
selection of variables.[6]
How do we create random trees? We create lots of random subsets from the
sample set of original yeast micro dataset array. We create decision trees from
these randomly chosen subsets.
For making decision trees, we enter training sets (X,y) where X is the 2D array
of all the inputs genes and their feature values and y is the array of features
of the corresponding target genes. [2] We plot the corresponding points of the
array on X-Axis and Y-Axis. We divide the X-Axis into several partitions and
each partition point forms decision node. We classify the dataset according to
the decision tree and for each partition or class we calculate the average mean
squared error based on the y values. Using mean squared error we nd the
weight from the decision tree. The weights obtained from several decision trees
are compared and the minimum value is chosen as the nal weight and entered
into the featureimportancesmatrix:
</p>
<h3></h3>
<p>
  These gene networks help in solving a numerous dierent biological and biomed-
ical problems. They are discussed below[1]:

Causal map of molecular interactions:
The most recent use found of gene regulatory networks is to act as a blueprint
of the molecular interactions. By this, a gene network can be used to derive
an ideal biological hypotheses about molecular interactions, such as for the
transcriptional regulation of genes experimented in wet labs. In such a condi-
tions GRNs represent causal bio-chemical interactions since the predicted links
are to correspond to actual physical binding events between the molecules. A
signicant side of this use is that the GRNs represent statistically signicant
predictions of potential molecular interactions which are obtained from very
large-scale data. Given a very large- scale number of potential molecular inter-
actions between 20,000 genes in Human beings and 6000 gene in organisms like
yeasts, the GRNs are of huge help in pruning these numbers down to potential
interactions for which statistical support is available. This, in turn enables more
eective and precise experiments.

Networks as biomarkers:
Recently, it has been debated that sub-networks can also be utilized as biomark
ers, for example for diagnostic and predictive and prognostic purposes. This
will be possible for any complex disorder such as cancer, as the hallmarks of
cancer genes are represented by pathways instead of individual genes. The
aspects of cancer pathway is that their constituting genes are continuously ac-
tively interacting with one another. Due to this network-based biomarkers can
be represented as statistical measures that will consider the molecular interac-
tion structure between individual genes. In contrary to biomarkers which are
based on individual genes, they neglect these completely Comparative network
analysis When multiple regulatory networks of dierent physiological and other
disease conditions will become available, we would be able to statistically com-
pare these networks. This will help us learn and understand about molecular
interaction changes across dierent physiological and other disease conditions
and enhance our biomedical understanding of such phenotypes. However, for
this approach to work it would be of necessity to create databases similar to the
protein structure databases, this will provide free access to the gene regulatory
networks from all dierent physiological and various diseased conditions.

Network medicine and drug design:
For structuring an ecient network medicine, it will be crucial to mix dierent
types of gene networks with one another other as each network type contains in-
formation regarding dierent molecular aspects. Hence for a better and ecient
design of drugs the use of gene networks is compulsory.
</p>

                    </div>
                </div>
                <div class="row" id="source">
                    <div class="col-lg-12" >
                        <h1>Source Code</h1>
                      <pre>  <code>
import pandas as p
import numpy as n
from scipy.io import loadmat
import math

from sklearn.ensemble import RandomForestRegressor
from sklearn.ensemble import RandomForestClassifier
from sklearn.svm import SVR
from sklearn.model_selection import cross_val_score
from sklearn import preprocessing
from sklearn import linear_model

# reading values of different genes of the yeast from the mat file
data_yeastnames = loadmat('ystnames.mat')

# reading values of the different features from the mat file
data_yeast_features = loadmat('yeast.mat')


# dataframe allows to create a multi dimensional matrix allowing to take combiations of rows and columns from different matrices

# here we are accessing the value of dictionary key called 'yystr' which is necessarily an one-D array
yeast_names = p.DataFrame(data_yeastnames['yystr'])

# here value of key 'Yeast' is accessed which consist of values of different features
yeast_features = p.DataFrame(data_yeast_features['Yeast'])


#we are creating a matrix with features of genes as columns and their names as rows
data_initial = p.DataFrame(data_yeast_features['Yeast'].T, columns=data_yeastnames['yystr'])


data_temp = data_initial


#number_genes consist of number of different genes of yeast data set which is 6178 in number in our case
number_genes = len(data_temp.keys())

network = p.DataFrame(n.zeros(shape=(len(data_temp.keys()), len(data_temp.keys()))), index=data_temp.keys(), columns=data_temp.keys())  #create 6178*6178 matrix with indexes of columns and rows as number oof yeast genes

#traversing i through values 0-6177
for i in range(number_genes):

    keys = list(data_temp.keys())

    #we are selecting each gene one by one to be targetGene gene
    targetGene = keys[i]
    keys.remove(targetGene)

    #del keys[2:number_genes]
    inputGenes = keys   #inut genes consist of all genes except the targetGene gene

    #print(targetGene)
    #print(inputGenes)

    # converting list to matrix
    X = data_temp[inputGenes]

    #converting list to matrix
    y = data_temp[targetGene]


    print('Training for targetGene gene {}: {}'.format(i, targetGene))

    #initialising random forest regression object with number of trees in the forest to be 200
    rfr = RandomForestRegressor(n_estimators=200, oob_score=True, n_jobs=-1)

    #this function builds the forest of 200 trees using inuts as X and Y
    rfr.fit(X, y)

    # print(clf)

    #feature_importances is 6187*6178 matrix consisiting the different weights with which the yeast genes are related to each other
    for i, x in zip(range(rfr.feature_importances_.size), rfr.feature_importances_):
         #print(i, inputGenes[i], x)

         #putting the values of weights into the grnmatrix
         network[targetGene][inputGenes[i]] = x


network.to_csv('yeast_grn.csv')  # putting the data to a file


                        </code></pre>

                    </div>
                </div>
                <div class="row" id="links">
                    <div class="col-lg-12" >
                        <h1>Video Links</h1>
                      <p>201401079 : Saloni Dalal - <a href="https://youtu.be/d0IbHsIWmzc">https://youtu.be/d0IbHsIWmzc</a></p>
                      <P>201401095 : Atrayee Deb - <a href="https://www.youtube.com/watch?v=vA_q3vEIR0w&feature=youtu.be"> https://www.youtube.com/watch?v=vA_q3vEIR0w&feature=youtu.be</a> </p>
                      <p>201401113 : Birva Patel - <a href="https://youtu.be/BmQ3qwSM7Y0"> https://youtu.be/BmQ3qwSM7Y0 </a></p>
                      <p>201401115 : Devanshi Bansal -<a href="https://youtu.be/LYihd_ysG-s"> https://youtu.be/LYihd_ysG-s</a> </p>
                      <p>201401159 : Ruchi Kachhia - <a href="https://youtu.be/7nX_nRiWJzo">https://youtu.be/7nX_nRiWJzo</a> </p>
                      <p>201401219 : Prachi Agarwal - <a href="https://youtu.be/aM5lQ3uwLsk" >https://youtu.be/aM5lQ3uwLsk</a> </p>

                    </div>
                </div>
            </div>
      </div>
    </div>
        <!-- /#page-content-wrapper -->

    <!-- jQuery -->
    <script src="js/jquery.js"></script>

    <!-- Bootstrap Core JavaScript -->
    <script src="js/bootstrap.min.js"></script>

    <!-- Menu Toggle Script -->
    <script>
    $("#menu-toggle").click(function(e) {
        e.preventDefault();
        $("#wrapper").toggleClass("toggled");
    });
    </script>
<style>
  .container{
    padding-right: 15px;
    padding-left: 0px;
    margin-right: auto;
    margin-left: 5px;
    margin-top: 5px;
  }
  #ppt{
    padding-top: 10px;
  }
</style>
</body>

</html>
